drivers/i2c/i2c_nrfx_twi.c

```diff
@@ -53,7 +53,11 @@ static int i2c_nrfx_twi_transfer(struct device *dev, struct i2c_msg *msgs,
 					       (msgs[i].flags & I2C_MSG_STOP) ?
 					       0 : NRFX_TWI_FLAG_TX_NO_STOP);
 		if (res != NRFX_SUCCESS) {
-			return -EIO;
+			if (res == NRFX_ERROR_BUSY) {
+				return -EBUSY;
```

Comment:
Not related to that PR, but that reminds me i²c API isn't very nice with the user as it does not provide the same thing as spi_context_lock() for concurrent access. (But both of the API do not provide timeout, which another issue).